Applications/Spire/Include/Spire/Ui/DropDownWindow.hpp

```diff
@@ -0,0 +1,46 @@
+#ifndef SPIRE_DROP_DOWN_WINDOW_HPP
+#define SPIRE_DROP_DOWN_WINDOW_HPP
+#include <QWidget>
+#include "Spire/Ui/DropShadow.hpp"
+
+namespace Spire {
+
+  //! Represents a pop-up window for displaying a widget.
+  class DropDownWindow : public QWidget {
+    public:
+
+      //! Constructs a DropDownWindow.
+      /*!
+        \param is_click_activated True if the DropDownWindow should be shown
+                                  or hidden in response to parent widget
+                                  mouse/keyboard activation events, false
+                                  otherwise.
+        \param parent The parent widget. Used to determine the position of the
+                      DropDownWindow.
+      */
+      explicit DropDownWindow(bool is_click_activated,
+        QWidget* parent = nullptr);
+
+      virtual ~DropDownWindow() = default;
+
+      //! Sets the widget to display in the DropDownWindow. Does nothing if a
```

Comment:
Try to reflect the "do nothing if" behavior in the method's name.

Reply:
Why does this method do nothing if the widget has been set? Invariants like these should be part of the constructor.

Reply:
If this method really can't reset the widget, then name it to ```initialize_widget```.

Reply:
I wanted the DropDownList to be a DropDownWindow, but in order to have the ScrollArea be the widget for the DropDownList I had to delay setting the widget on the base class DropDownWindow until it's created by the DropDownList.

I was trying to avoid composing objects in certain cases like I mentioned in the case notes (issues with filters and layouts), but I think in this case I'll modify this to have the DropDownList just inherit from QWidget.